src/java.base/share/classes/java/lang/String.java

```diff
@@ -3219,9 +3286,9 @@ public String replace(CharSequence target, CharSequence replacement) {
      *
      * @see java.util.regex.Pattern
      *
-     * @since 1.4
+     * @since 21
      */
-    public String[] split(String regex, int limit) {
+    public String[] split(String regex, int limit, boolean withDelimiters) {
```

Comment:
Is it really a suitable design to use a flag argument to decide whether to include delimiters?

Using an argument is usually good when the caller might want to decide whether to do something based on a variable, but that is probably uncommon for this feature.

I this case I think a separate method name would be more clear for user:

```
// Easy to understand
s.splintWithDelimiters("pattern")
```
Instead of:

```
// Harder to understand
s.splint("pattern", true)
```